pkg/operator/controller/controller_lb.go

```diff
@@ -34,8 +34,12 @@ const (
 // ensureLoadBalancerService creates an LB service if one is desired but absent.
 // Always returns the current LB service if one exists (whether it already
 // existed or was created during the course of the function).
-func (r *reconciler) ensureLoadBalancerService(ci *ingressv1alpha1.ClusterIngress, deployment *appsv1.Deployment, infraConfig *configv1.Infrastructure) (*corev1.Service, error) {
-	desiredLBService, err := desiredLoadBalancerService(ci, deployment, infraConfig)
+func (r *reconciler) ensureLoadBalancerService(ci *ingressv1alpha1.ClusterIngress, deployment *appsv1.Deployment, infraConfig *configv1.Infrastructure, ha *ingressv1alpha1.ClusterIngressHighAvailability) (*corev1.Service, error) {
+	if deployment == nil || ha == nil {
```

Comment:
Remove ha pointer and check whether the type is Cloud?

Reply:
You mean lift the check out of `desiredLoadBalancerService`? That would work in this case, but in the general pattern, an ensure method might need to do some clean-up—for example, `ensureDefaultCertificateForIngress` exits immediately if `deployment == nil` because it cannot do anything in that case, but it does not exit immediately if `domain == nil` because that case indicates that the desired secret is nil, meaning the ensure method must delete any current secret.

The pattern I have been following is that the ensure method returns immediately if any prerequisite values are missing, and the desired function will return nil if no resource should exist.  In the case of `ensureLoadBalancerService`, since it only ever creates and never deletes or updates, it happens to be the case that returning nil from the desired function has the same effect as immediately returning nil from the ensure method would have.

But will `ensureLoadBalancerService` handle clean-up in the future? Do we want to follow the pattern demonstrated by `ensureDefaultCertificateForIngress`?

Reply:
Removed the pointer check.